src/Storefront/Framework/Routing/RequestTransformer.php

```diff
@@ -100,14 +100,15 @@ public function transform(Request $request): Request
         $request = $this->decorated->transform($request);
 
         if (!$this->isSalesChannelRequired($request->getPathInfo())) {
-            return $this->decorated->transform($request);
```

Comment:
@mitelg Do you know why we have to call $this->decorated->transform twice (Line 100 & 103)? is it intended?